docs/rules/require-tagless-components.md

```diff
@@ -0,0 +1,66 @@
+## Components must not have a wrapper element
+
+### Rule name: `require-tagless-components`
+
+Ember allows you to disable the wrapper element on a component (turning it into a "tagless" component). This is now the preferred style and the _only_ style allowed with Glimmer components.
+
+Instead of having the wrapper element implicitly defined by the component, all DOM elements should be represented in the component's template.
+
+#### Examples of **incorrect** code for this rule:
+
+```javascript
+import Component from '@ember/component';
+
+export default Component.extend({});
+```
+
+```javascript
+import Component from '@ember/component';
+
+export default Component.extend({
+  tagName: 'span'
+});
+```
+
+```javascript
+import Component from '@ember/component';
+
+export default class MyComponent extends Component {}
+```
+
+```javascript
+import Component from '@ember/component';
+
+export default class MyComponent extends Component {
+  tagName = 'span'
+}
+```
+
+#### Examples of **correct** code for this rule:
+
+```javascript
+import Component from '@ember/component';
+
+export default Component.extend({
+  tagName: ''
+});
+```
+
+```javascript
+import Component from '@ember/component';
+
+export default class MyComponent extends Component {
+  tagName = ''
+}
+```
+
+```javascript
+import Component from '@glimmer/component';
+
+export default class MyComponent extends Component {}
+```
+
+#### Caveats
+
+* Rule does not catch cases where a Mixin is included to configure the `tagName` property.
+* Rule does not catch cases where a decorator is applied to configure the `tagName` property.
```

Comment:
Can you add a `References` section at the bottom which links to relevant documentation?

Reply:
For sure -- I've been looking around and, while I know that tag-less are the recommended approach, I haven't actually been able to find anywhere that spells that out explicitly.

What do you think about linking to information about Glimmer components, and in the References section, mentioning that using Ember components in tag-less mode is recommended to aid the transition to Glimmer components?

Reply:
Sounds good.